### PR TITLE
Add a tutorial on porting test groups from legacy

### PR DESCRIPTION
**Pull request by xylar**
This tutorial uses the `gotm` test group and the default test case as an example.

### CODE REVIEW

**Comment by xylar** [comment]
@mark-petersen and @matthewhoffman, in the spirit of being ready for new students, I made the first of what I expect to be several tutorials in the compass documentation.  This one is on porting a legacy test group.  Given that it took me 3 days (off an on) to put together, these tutorials aren't going to come all that quickly.  But maybe this is one of the more critical?

Anyway, I appreciate any feedback.  The easiest way is if you actually edit the rst file using the "suggest" feature in GitHub.   Highlight the relevant lines and, in your comment, hit the suggest button then edit the text. That way, I can aggregate suggested changes here and make a commit.

**Comment by xylar** [comment]
@qingli411, it would also be helpful to have your review on this one, and reading this tutorial might give you a better handle on reviewing #129

**Comment by matthewhoffman** [comment]
@xylar , can you remind me how to build the documentation in this branch for easier reading?

**Comment by xylar** [comment]
Sure:
```
source test_compass_1.0.0*.sh
cd docs
make html
google-chrome _build/html/index.html
```

**Comment by xylar** [comment]
@vanroekel ran into trouble with the `conda/configure_compass_env.sh` in the SCORPIO build on his Mac (and I saw it, too).  I'm working on that as I have time but in the meantime you might need to copy the `_build/html` directory over to somewhere that you can open it in a browser.

**Review by mark-petersen — APPROVED** [review]
@xylar I looked over the rst files. It looks good. Thanks for adding this documentation.

File: docs/tutorials/dev_porting_legacy.rst [hunk 1]
```diff
@@ -0,0 +1,1296 @@
...
+``subprocess``` call to ``MpasCellCuller.x``).  ``cull()`` has
+:py:class:`xarray.Dataset` objects as its input an return value, and also
+takes a "logger" where it can write output (sometimes a log file and sometimes
+directly to the terminal via stdout).  You should always pass ``self.logger``
+so ``compass`` can figure out whether a file or stdout is the right place for
+output to go.
+
+.. note::
+
+    In this particular test case, the there are no land cells defined and
```
Comment (qingli411) [inline docs/tutorials/dev_porting_legacy.rst:758]:
```suggestion
    In this particular test case, there are no land cells defined and
```

File: docs/tutorials/dev_porting_legacy.rst [hunk 2]
```diff
@@ -0,0 +1,1296 @@
...
+``compass run`` within the work directory.  See :ref:`dev_command_line` for
+more on that process.
+
+Documentation
+-------------
+
+Make sure to add some documentation of your new test group.  You need to add
+all of the functions, classes and methods to the API documentation in
+``docs/developers_guide/<core>/api.rst``, following the exmaples for other
+test groups.  You also need to add a file to both the user's guid and the
```
Comment (qingli411) [inline docs/tutorials/dev_porting_legacy.rst:1272]:
```suggestion
test groups.  You also need to add a file to both the user's guide and the
```

**Review by qingli411 — APPROVED** [review]
@xylar This tutorial is really helpful! Thanks for creating it! It reads very well and is very informative. I only find a few typos which are highlighted in the suggested changes.

**Comment by xylar** [comment]
@qingli411, thanks for fixing so many typos! I wish I were surprised. But I know that's just how it goes for me.

**Comment by xylar** [comment]
I updated the tutorial following the fix in #144.  I also fixed a bunch more typos and replaced all tabs with spaces (even when there are tabs in the legacy COMPASS files).

**Comment by xylar** [comment]
@matthewhoffman, do you have a sense of when you might have time to review this?  It's not a big rush but I also don't want it to get forgotten.

**Comment by matthewhoffman** [comment]
I've had it sitting open in a tab for the past week.  I'll try to buckle down and look at it today or tomorrow.

File: docs/tutorials/dev_porting_legacy.rst [hunk 3]
```diff
@@ -0,0 +1,1295 @@
...
+    <immutable_stream name="input_init"
+                      type="input"
+                      filename_template="mesh.nc"
+                      input_interval="initial_only"/>
+
+Defining config options
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The remainder of the ``init`` step will consist of defining the ``run()``
+method that does the refal work of the step.  To make it easier for users to
```
Comment (matthewhoffman) [inline docs/tutorials/dev_porting_legacy.rst:634]: refal typo

File: docs/tutorials/dev_porting_legacy.rst [hunk 4]
```diff
@@ -0,0 +1,1295 @@
...
+    # the depth of the sea floor (m)
+    bottom_depth = 15.0
+
+By default, the domain is 4 x 4 horizontal cells, each 2.5 km in size.  The
+ocean is 15 m deep, divided over 250 uniformly spaced levels.  A user could
+change any of these parameters before running the ``init`` step to modify
+the initial condition (and therefore the rest of the test case).
+
+Since the config file has the same name (``default``) as the test case, it will
+be included automatically when the config file produced when test case gets
```
Comment (matthewhoffman) [inline docs/tutorials/dev_porting_legacy.rst:665]:
```suggestion
be included automatically when the config file is produced when the test case gets
```

File: docs/tutorials/dev_porting_legacy.rst [hunk 5]
```diff
@@ -0,0 +1,1295 @@
...
+        <run_script name="run.py">
+            <step executable="./plot_profile.py">
+            </step>
+        </run_script>
+    </config>
+
+Symlinks are created to a plotting script and the output from the ``forward``
+step, and then the plot script is run.
+
+An identical approach could be used in ``compass`` but it is not the perferred
```
Comment (matthewhoffman) [inline docs/tutorials/dev_porting_legacy.rst:1046]:
```suggestion
An identical approach could be used in ``compass`` but it is not the preferred
```

File: docs/tutorials/dev_porting_legacy.rst [hunk 6]
```diff
@@ -0,0 +1,1295 @@
...
+
+Make sure to add some documentation of your new test group.  You need to add
+all of the functions, classes and methods to the API documentation in
+``docs/developers_guide/<core>/api.rst``, following the examples for other
+test groups.  You also need to add a file to both the user's guide and the
+developer's guide describing the test group and its test cases and steps.
+
+For the user's guide, create a file
+``docs/users_guide/<core>/test_groups/<test_group>.rst``.  In that file, you
+should describe what the test group and its test cases do in a way that would
```
Comment (matthewhoffman) [inline docs/tutorials/dev_porting_legacy.rst:1276]:
```suggestion
should describe what the test group and what its test cases do in a way that would
```

**Review by matthewhoffman — APPROVED** [review]
This is a very useful tutorial.  I think it will be helpful to developers creating a new test case, even if they haven't used legacy COMPASS.

File: docs/tutorials/dev_porting_legacy.rst [hunk 7]
```diff
@@ -0,0 +1,1295 @@
...
+you are working on a "supported" machine and using the default compilers and
+MPI libraries, but consult the documentation to make an environment to suit
+your needs.
+
+.. code-block:: bash
+
+  # this one will take a while the first time
+  ./conda/create_compass_env.py --conda $HOME/miniconda3
+
+If all goes well, you will have a file named ``test_compass_1.0.0*.sh``, where
```
[the remark this answers is not in the document]
Comment (xylar) [reply docs/tutorials/dev_porting_legacy.rst:46]:
```suggestion
If all goes well, you will have a file named ``load_dev_compass_1.0.0*.sh``, where
```

File: docs/tutorials/dev_porting_legacy.rst [hunk 8]
```diff
@@ -0,0 +1,1295 @@
...
+and a python script for plotting the results compared to analytic solutions
+(``plot_profile.py``).
+
+Making a new test group
+-----------------------
+
+Okay, with those details as a reference point from legacy COMPASS, let's jump
+into developing the new test group in ``compass``.  Use any method you like
+for editing code.  If you haven't settled on a method and are working on your
+own laptop or desktop, you may want to try and integrated development
```
[the remark this answers is not in the document]
Comment (xylar) [reply docs/tutorials/dev_porting_legacy.rst:177]:
```suggestion
own laptop or desktop, you may want to try an integrated development
```

File: docs/tutorials/dev_porting_legacy.rst [hunk 9]
```diff
@@ -0,0 +1,1295 @@
...
+Next, let's make a new class for the ``gotm`` test group in ``__init__.py``:
+
+.. code-block:: python
+
+    from compass.testgroup import TestGroup
+
+
+    class Gotm(TestGroup):
+        """
+        A test group for baroclinic channel test cases
```
[the remark this answers is not in the document]
Comment (xylar) [reply docs/tutorials/dev_porting_legacy.rst:201]:
```suggestion
        A test group for General Ocean Turbulence Model (GOTM) test cases
```

File: docs/tutorials/dev_porting_legacy.rst (hunk 3 above)
Comment (xylar) [reply docs/tutorials/dev_porting_legacy.rst:634]:
```suggestion
method that does the real work of the step.  To make it easier for users to
```

**Comment by xylar** [comment]
@matthewhoffman, @qingli411 and @mark-petersen, thanks for the reviews!  I'll merge this as soon as the tests have finished.

**Pull request merged**